### PR TITLE
FieldConfig: implement color picker

### PR DESCRIPTION
**Pull request by ryantxu**
The current implementation for color picker just says `<div>todo</div>`, for the [clock-panel](https://github.com/grafana/clock-panel/blob/master/src/options.tsx#L25), I hacked something together, but it should be better supported in core.

image
image

TODO:
- [x] clicking on the body should open the popup

### CODE REVIEW

**Comment by torkelo** [comment]
But I opened this already in this weekend with the color mode options?

**Comment by torkelo** [comment]
The color standard option is not only a color picker, it is the mode as well , which was what I opened a draft PR for

**Comment by torkelo** [comment]
https://github.com/grafana/grafana/pull/24763

**Comment by torkelo** [comment]
But this maybe is not about the field config color option editor but a general standard editor for a color picker option?

**Comment by ryantxu** [comment]
Intersting -- ya #24763 uses the same component, but does something else.  The intention here is really to have something that picks a color.  In #24763 you are using the same field for something more like picking a "color mode"

File: public/app/plugins/panel/text2/module.tsx [hunk 1]
```diff
@@ -19,6 +19,16 @@ export const plugin = new PanelPlugin<TextOptions>(TextPanel)
...
+      .addColorPicker({
+        path: 'color',
+        name: 'Color',
+        description: 'just a test... no color pickers in core?',
+        settings: {
+          defaultText: 'Pick background',
+          disableNamedColors: false,
+          allowUndefined: true,
+        },
+      })
```
Comment (ryantxu) [inline public/app/plugins/panel/text2/module.tsx:31]: This should not be commited!  but no story or core component that wants color?

Comment (dprokop) [reply public/app/plugins/panel/text2/module.tsx:31]: Yes we need color picker, just haven't migrated any piece of code it was necessary :)

**Comment by ryantxu** [comment]
@dprokop -- this is ready to go, and is for sure better than "TODO" :)  would be nice to have a color picker available for plugin configs.

Since it is replacing a TODO -- should this be 7.0.3?

File: packages/grafana-ui/src/components/OptionsUI/color.tsx [hunk 2]
```diff
@@ -1,11 +1,80 @@
...
+import { Icon } from '../Icon/Icon';
+import { css } from 'emotion';
+import { ColorPickerTrigger } from '../ColorPicker/ColorPickerTrigger';
 
 export const ColorValueEditor: React.FC<FieldConfigEditorProps<string, ColorFieldConfigSettings>> = ({
   value,
   onChange,
   item,
 }) => {
-  return <div>todo</div>;
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/OptionsUI/color.tsx:10]: 7.0 behavior :(

**Review by torkelo — APPROVED** [review]
Looks good, 7.0.3 worthy

**Pull request merged**